modules/color.lua

```diff
@@ -23,6 +25,14 @@ exports.set_color = function (fore, back, bold)
     return "\x1b[3"..fore..";"..bold..";".."4"..back.."m"
 end
 
+exports.get_clink_color = function (setting_name)
+    local sgr = ver.supports_color_settings() and settings.get(setting_name) or ""
```

Comment:
I wonder what would `settings.get("color.git.star")` return? Is it single color number from [this table](https://en.wikipedia.org/wiki/ANSI_escape_code#SGR_parameters) or complete EGR-sequence or something  else? I assume it should be either 1) or 2) but just wanted to make sure.

Also in case it's a single EGR (color) code,  should there be a separate function to convert color code to full escape sequence? Can't you simply do something like `colors.color_text("foo", settings.get("color.git.star"))` instead?

Reply:
It's a complete SGR Parameters sequence, and can include anything the user wants to include, even 24-bit color or underline and so on.

My concern about `colors.color_text("foo", settings.get("color.git.star"))` is that it doesn't encapsulate the version check.  The caller must wrap it with their own version check to avoid blowing up.  If Clink doesn't support color settings, I would like for that to be the same as the user having set the color setting to nothing, and just return the empty string.  Then the caller doesn't have to add a version check -- which is both a convenience and a defensive design: if the caller forgets to check the version, their code will still work correctly.

I'm happy to rename `color.get_clink_color` if another name is preferred, but I think its functional contract is the right one to have, and I don't want to change that part.

Reply:
Nah, since `settings.get` would return a complete EGR sequence `color_text` wouldn't work - I just feel like there could be a better/more descriptive/less specific name rather than `get_clink_color` but I can't come up with the better option ATM os let's leave it as is